haystack/nodes/prompt/invocation_layer/anthropic_claude.py

```diff
@@ -0,0 +1,202 @@
+        self.max_length = max_length
+
+        # Due to reflective construction of all invocation layers we might receive some
+        # unknown kwargs, so we need to take only the relevant.
+        # For more details refer to Anthropic documentation
+        supported_kwargs = ["temperature", "top_p", "top_k", "stop_sequences", "stream", "stream_handler"]
+        self.model_input_kwargs = {k: v for (k, v) in kwargs.items() if k in supported_kwargs}
+
+        # Number of max tokens is based on the official Anthropic model pricing from:
+        # https://cdn2.assets-servd.host/anthropic-website/production/images/FINAL-PRICING.pdf
+        self.max_tokens_limit = 9000
+        self.tokenizer: GPT2TokenizerFast = GPT2TokenizerFast.from_pretrained("gpt2")
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes a prompt on the model. It takes in a prompt and returns a list of responses using a REST invocation.
+        :return: The responses are being returned.
+        """
+
+        human_prompt = "\n\nHuman: "
+        assitant_prompt = "\n\nAssistant: "
+
+        prompt = kwargs.get("prompt")
+        if not prompt:
+            raise ValueError(
+                f"No prompt provided. Model {self.model_name_or_path} requires prompt."
+                f"Make sure to provide prompt in kwargs."
+            )
+
+        kwargs_with_defaults = self.model_input_kwargs
+
+        # we use keyword stop_words but Anthropic uses stop_sequences
+        if "stop_words" in kwargs:
+            kwargs["stop_sequences"] = kwargs.pop("stop_words")
+        if "max_tokens" in kwargs:
+            kwargs["max_length"] = kwargs.pop("max_length")
+        kwargs_with_defaults.update(kwargs)
+
+        # either stream is True (will use default handler) or stream_handler is provided
+        stream = (
+            kwargs_with_defaults.get("stream", False) or kwargs_with_defaults.get("stream_handler", None) is not None
+        )
+        stop_sequences = kwargs.get("stop_sequences", ["\n\nHuman: "])
+
+        data = {
+            "model": self.model_name_or_path,
+            "prompt": "{} {} {}".format(human_prompt, prompt, assitant_prompt),
+            "max_tokens_to_sample": kwargs_with_defaults.get("max_length", self.max_length),
+            "temperature": kwargs_with_defaults.get("temperature", 1),
+            "top_p": kwargs_with_defaults.get("top_p", -1),
+            "top_k": kwargs_with_defaults.get("top_k", -1),
+            "stream": stream,
+            "stop_sequences": stop_sequences,
+        }
+
+        if not stream:
+            res = self._post(data=data)
+            return [res.json()["completion"].strip()]
+
+        res = self._post(data=data, stream=True)
+        handler: TokenStreamingHandler = kwargs_with_defaults.pop("stream_handler", DefaultTokenStreamingHandler())
+        client = sseclient.SSEClient(res)
+        tokens: List[str] = []
+        try:
+            for event in client.events():
```

Comment:
@silvanocerza does this actually work? It is the streaming format from OpenAI. Are they using the same format?